xbmc/guilib/GUIEditControl.cpp

```diff
@@ -424,45 +447,55 @@ void CGUIEditControl::ProcessText(unsigned int currentTime)
   if (m_smsTimer.IsRunning() && m_smsTimer.GetElapsedMilliseconds() > smsDelay)
     UpdateText();
 
-  if (m_bInvalidated)
-  {
-    m_label.SetMaxRect(m_posX, m_posY, m_width, m_height);
-    m_label.SetText(m_info.GetLabel(GetParentID()));
-    RecalcLabelPosition();
-  }
-
   bool changed = false;
+  changed |= m_label.SetText(m_info.GetLabel(m_parentID));
 
-  m_clipRect.x1 = m_label.GetRenderRect().x1;
-  m_clipRect.x2 = m_clipRect.x1 + m_label.GetMaxWidth();
+  m_clipRect.x1 = m_posX + m_label.GetLabelInfo().offsetX;
+  m_clipRect.x2 = m_clipRect.x1 + m_width - 2 * m_label.GetLabelInfo().offsetX;
   m_clipRect.y1 = m_posY;
   m_clipRect.y2 = m_posY + m_height;
 
-  // start by rendering the normal text
-  float leftTextWidth = m_label.GetRenderRect().Width();
+  // Limit left text max width to 50% of space when focused, otherwise 70%
+  const float maxTextWidth = m_width * (HasFocus() ? 0.5f : 0.7f);
+
+  const float leftTextWidth =
+      std::min(m_label.GetTextWidth(), maxTextWidth - 2 * m_label.GetLabelInfo().offsetX);
+
+  changed |= m_label.SetMaxRect(m_posX, m_posY, maxTextWidth, m_height);
+
+  if (m_bInvalidated)
+  {
+    if (!HasFocus() && leftTextWidth > 0)
+      m_textOffset = 0;
+    else
+      RecalcRightLabelPosition();
+  }
+
   if (leftTextWidth > 0)
   {
     // render the text on the left
+    changed |= m_label.SetScrolling(HasFocus());
     changed |= m_label.SetColor(GetTextColor());
     changed |= m_label.Process(currentTime);
 
-    m_clipRect.x1 += leftTextWidth + spaceWidth;
+    m_clipRect.x1 += leftTextWidth + TEXT_SPACE;
   }
 
+  // render the text on the right
+
   if (CServiceBroker::GetWinSystem()->GetGfxContext().SetClipRegion(m_clipRect.x1, m_clipRect.y1, m_clipRect.Width(), m_clipRect.Height()))
   {
+    // set alignment for right label text
     uint32_t align = m_label.GetLabelInfo().align & XBFONT_CENTER_Y; // start aligned left
-    if (m_label2.GetTextWidth() < m_clipRect.Width())
-    { // align text as our text fits
-      if (leftTextWidth > 0)
-      { // right align as we have 2 labels
-        align |= XBFONT_RIGHT;
-      }
-      else
-      { // align by whatever the skinner requests
-        align |= (m_label2.GetLabelInfo().align & 3);
-      }
+    if (leftTextWidth > 0)
+    { // right align as we have 2 labels
+      align |= XBFONT_RIGHT;
+    }
+    else
+    { // align by whatever the skinner requests
+      align |= (m_label2.GetLabelInfo().align & 3);
```

Comment:
(XBFONT_RIGHT | XBFONT_CENTER_X) instead of 3?
horizontal alignment XBFONT_JUSTIFIED is ignored, on purpose?

Reply:
no idea this is not my code then i have no investigated
i have just changed space indentation
cleanups on existing code can be done with separate prs

Reply:
I suppose justified wouldn't be very useful in an edit control.
Replacing the magic number would still be nice, did you have other things in mind for a cleanup PR?

Reply:
i have no plan to make a cleanup pr soon...
i have add a separate commit to replace magic n